app/workers/recover_lost_jobs_worker.rb

```diff
@@ -15,6 +17,23 @@ def old_unprocessed(scope)
     scope.where(processed_at: nil).where("created_at <= ?", 1.hour.ago)
   end
 
+  def non_existent_daily_digests
+    expected_digest_week.map { |date| DigestRun.new(date: date, range: :daily) }
+      .reject { |digest_run| DigestRun.exists?(date: digest_run.date, range: :daily) }
+  end
+
+  def non_existent_weekly_digests
+    [DigestRun.new(date: expected_digest_week.find(&:saturday?), range: :weekly)]
+      .reject { |digest_run| DigestRun.exists?(date: digest_run.date, range: :weekly) }
+  end
+
+  def expected_digest_week
+    digestion_time = Time.zone.parse("#{DigestRun::DIGEST_RANGE_HOUR}:00")
+    cutoff_with_delay = digestion_time + 1.hour
+    end_date = Time.zone.now > cutoff_with_delay ? Time.zone.today : Time.zone.yesterday
```

Comment:
Hmm are we intending to wait an hour or 30 minutes before trying to create a missing digest run?

I notice that `DigestRun::DIGEST_RANGE_HOUR` is set for 8am but we don't actually start any digests until [8:30am]. This would mean the the `cutoff_with_delay` would be 9am leaving a 30 minute window for the digest run to be created which should be fine but just checking that was intended.

[8:30am]: https://github.com/alphagov/email-alert-api/blob/master/config/sidekiq.yml#L18-L23

Reply:
Haha, no it wasn't! I completely missed the half hour discrepancy. However, I do think it's fine: half an hour is still plenty of time, and I think it's actually a good thing that the recovery is ignorant of the original scheduling: we really _should_ have a new digest run if we're one hour in to a new digest day.

> Hmm are we intending to wait an hour or 30 minutes before trying to create a missing digest run?

In conclusion: kind of both 😄.

Reply:
Yeah `DigestRun::DIGEST_RANGE_HOUR` threw me a bit that it wasn't the same as our actual scheduled digest run time but hey, it does say just 'hour' in the name. 

Half an hour seems fine! 👍